src/preamble.js

```diff
@@ -932,6 +932,13 @@ function stackTrace() {
 }
 {{{ maybeExport('stackTrace') }}}
 
+function isAbsoluteUrl(url) {
+  return /^(?:[a-z]+:)?\/\//i.test(url);
+}
+
+function joinUrl(a, b) {
+  return a && !isAbsoluteUrl(b) ? a + b : b;
```

Comment:
this seems to assume that a ends with `/`, perhaps in ASSERTIONS we should check that?

Reply:
Good point, added.